### PR TITLE
Fix error for component name with :

### LINKED ISSUE #803: [TwigComponent] namespace double dot isn't parsed correctly
First of all, thanks for this awesome component!

I noticed that create a twig component using a namespace wasn't working as expected using some syntaxes.

Let's assume a component named `foo:bar`.
Using quotes, it works as expected:

```twig
{% component 'foo:bar' %}
  {# ... #}
{% endcomponent %}
```

But if I try the same without quotes or using the html syntax:

```twig
{% component foo:bar %}
  {# ... #}
{% endcomponent %}

{# OR #}

<twig:foo:bar>
  {# ... #}
</twig:foo:bar>
```

This throw a `SyntaxError`:
```
An exception has been thrown during the compilation of a template ("Unknown component "foo". The registered components are: foo:bar").
```

As you can see, it is only taking in account the namespace as component name.

### PR DESCRIPTION
**Pull request by WebMamba**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Tickets       | Fix #803 
| License       | MIT

### CODE REVIEW

File: src/TwigComponent/src/Twig/TwigPreLexer.php [hunk 1]
```diff
@@ -117,7 +117,7 @@ public function preLexComponents(string $input): string
     private function consumeComponentName(): string
     {
         $start = $this->position;
-        while ($this->position < $this->length && preg_match('/[A-Za-z0-9_]/', $this->input[$this->position])) {
+        while ($this->position < $this->length && preg_match('/[A-Za-z0-9_:]/', $this->input[$this->position])) {
```
Comment (WebMamba) [inline src/TwigComponent/src/Twig/TwigPreLexer.php]: Do you think about other characters that should be allowed here?

**Comment by WebMamba** [comment]
This is a fix on the HTML twig component syntax but do you think we should extend the fix to allow this
`` {% component foo:bar `` for the moment only ``{% component 'foo:bar' %}`` is allow?

File: src/TwigComponent/src/Twig/TwigPreLexer.php [hunk 2]
```diff
@@ -58,7 +58,7 @@ public function preLexComponents(string $input): string
                     $this->currentComponents[] = ['name' => $componentName, 'hasDefaultBlock' => false];
                 }
 
-                $output .= "{% component {$componentName}".($attributes ? " with { {$attributes} }" : '').' %}';
+                $output .= "{% component '$componentName'".($attributes ? " with { {$attributes} }" : '').' %}';
```
Comment (jdreesen) [inline src/TwigComponent/src/Twig/TwigPreLexer.php]:
```suggestion
                $output .= "{% component '{$componentName}'".($attributes ? " with { {$attributes} }" : '').' %}';
```

Comment (WebMamba) [reply src/TwigComponent/src/Twig/TwigPreLexer.php]: good catch thanks

File: src/TwigComponent/src/Twig/TwigPreLexer.php (hunk 1 above)
Comment (kbond) [inline src/TwigComponent/src/Twig/TwigPreLexer.php]: Should we future proof this by allowing `-`, `/`, `@` and `.`?

Comment (WebMamba) [reply src/TwigComponent/src/Twig/TwigPreLexer.php]: and - ?

Comment (kbond) [reply src/TwigComponent/src/Twig/TwigPreLexer.php]: `<twig:my-component/>`?

Comment (WebMamba) [reply src/TwigComponent/src/Twig/TwigPreLexer.php]: yeah why not! We never know what people can bring 😅

Comment (WebMamba) [reply src/TwigComponent/src/Twig/TwigPreLexer.php]: ha ok fine that was the first on your list! I didn't see it! Sorry about that

Comment (kbond) [reply src/TwigComponent/src/Twig/TwigPreLexer.php]: (I edited the comment) 😁

File: src/TwigComponent/tests/Unit/TwigPreLexerTest.php [hunk 3]
```diff
@@ -29,47 +29,47 @@ public function getLexTests(): iterable
...
             '<twig:foo>Foo</twig:foo>',
-            '{% component foo %}{% block content %}Foo{% endblock %}{% endcomponent %}',
+            '{% component \'foo\' %}{% block content %}Foo{% endblock %}{% endcomponent %}',
         ];
 
         yield 'component_with_default_block_that_holds_a_component_and_multi_blocks' => [
             '<twig:foo>Foo <twig:bar /><twig:block name="other_block">Other block</twig:block></twig:foo>',
-            '{% component foo %}{% block content %}Foo {% component bar %}{% endcomponent %}{% endblock %}{% block other_block %}Other block{% endblock %}{% endcomponent %}',
+            '{% component \'foo\' %}{% block content %}Foo {% component \'bar\' %}{% endcomponent %}{% endblock %}{% block other_block %}Other block{% endblock %}{% endcomponent %}',
         ];
```
Comment (weaverryan) [inline src/TwigComponent/tests/Unit/TwigPreLexerTest.php:73]: Let's add tests specifically for any of the new character we allow - e.g. `<twig:foo:bar` or `<twig:foo-bar`, etc

**Review by weaverryan — COMMENTED** [review]
Thanks for getting this rolling ❤️

File: src/TwigComponent/src/Twig/TwigPreLexer.php [hunk 4]
```diff
@@ -58,7 +58,7 @@ public function preLexComponents(string $input): string
                     $this->currentComponents[] = ['name' => $componentName, 'hasDefaultBlock' => false];
                 }
 
-                $output .= "{% component {$componentName}".($attributes ? " with { {$attributes} }" : '').' %}';
+                $output .= "{% component '{$componentName}'".($attributes ? " with { {$attributes} }" : '').' %}';
```
Comment (kbond) [inline src/TwigComponent/src/Twig/TwigPreLexer.php:61]: Does `{% component @foo/bar-baz.html.twig %}` (no `'`) not work?

Comment (WebMamba) [reply src/TwigComponent/src/Twig/TwigPreLexer.php:61]: Yes It works with both

**Review by kbond — APPROVED** [review]

**Comment by weaverryan** [comment]
Thanks Matheo!

**Comment by Jupi007** [comment]
Thanks a lot for the fix @WebMamba !

**Pull request merged**